src/storm-pomdp/parser/PomdpSolveParser.cpp

```diff
@@ -0,0 +1,422 @@
+                std::getline(iss, action, ':');  // Action
+                action = trim(action); 
+                if (iss.peek() == ':') {
+                    // Single Transition Probability (T: <action> : <start-state> : <end-state> %f)
+                    std::getline(iss, start_state, ':');
+                    std::getline(iss, end_state);
+                    double probability;
+                    iss >> probability;
+
+                    start_state = trim(start_state);
+                    end_state = trim(end_state);
+
+                    if (probability != 0) {
+                        if (action == "*") {
+                            for (const auto& act : pomdp.actions) {
+                                pomdp.transitions[act][start_state + ":" + end_state] = probability;
+                            }
+                        } else if (start_state == "*") {
+                            for (const auto& state : pomdp.states) {
+                                pomdp.transitions[action][state + ":" + end_state] = probability;
+                            }
+                        } else {
+                            pomdp.transitions[action][start_state + ":" + end_state] = probability;
+                        }
+                    }
+                } else {
+                    // Single Row of Transition Matrix (T: <action> : <start-state> %f %f ... %f)
+                    std::getline(iss, action, ':');
+                    action = trim(action);
+                    iss >> start_state;
+                    start_state = trim(start_state);
+                    double prob;
+                    while (iss >> prob) {
+                        probabilities.push_back(prob);
+                    }
+
+                    for (int i = 0; i < pomdp.states.size(); i++) {
+                        if (probabilities[i] != 0) {
+                            if (action == "*") {
+                                for (const auto& act : pomdp.actions) {
+                                    pomdp.transitions[act][start_state + ":" + pomdp.states[i]] = probabilities[i];
+                                }
+                            } else {
+                                pomdp.transitions[action][start_state + ":" + pomdp.states[i]] = probabilities[i];
+                            }
+                        }
+                    }
+                }
+            } else {
+                // Entire Transition Matrix for an Action (T: <action>)
+                action = trim(line.substr(3)); // Get action from "T: <action>"
+
+                for (int i = 0; i < pomdp.states.size(); i++) {
+                    std::getline(infile, line); // Read each row
+                    std::istringstream rowStream(line);
+                    for (int j = 0; j < pomdp.states.size(); j++) {
+                        double prob;
+                        rowStream >> prob;
+                        if (prob != 0) {
```

Comment:
I'm not exactly sure what the semantics of the ``>>`` operator are if you use it to fill a double. Can it be that we get a floating point error such that ``prob`` is not 0, even though it should be?